### PR TITLE
doxygen: Refactoring the doxygen book generator

### PR DESCRIPTION
**Pull request by leemgs**
Fixed issue #554.

This commit is to refactor the existing doxygen generator to improve
readability and maintenance.

* before: book-hard-copy-prj-generate.sh
* after : book-doxgen-publish.sh

**Changelog**
  * Added doxybook newly for maintenance
  * Replaced book-hard-copy-prj-generator.sh with book-doxygen-publish.sh
  * Added generate_clean function to remove unnecessary files
  * Updated the time format layout
  * Refactored the existing doxygen geneation functions

Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>

---

### CODE REVIEW

File: ci/doc/how-to-setup-taos-ci-server.md [hunk 1]
```diff
@@ -216,9 +216,9 @@ Finally, let's generate automatically PDF book per 1 hour with cron table (e.g.,
 $ sudo vi /etc/crontab
 # Generate doxygen document
 20 * * * * www-data cd /var/www/html/{your_prj_name}/ ; git pull
-30 * * * * www-data /var/www/html/{your_prj_name}/doc/book-hard-copy-prj-generate.sh
+30 * * * * www-data /var/www/html/{your_prj_name}/doc/book-doxygen-publish.sh
```
Comment (dongju-chae) [inline ci/doc/how-to-setup-taos-ci-server.md]: Is it now `../doxybook/..` instead of `../doc/..`?

Comment (dongju-chae) [reply ci/doc/how-to-setup-taos-ci-server.md]: Ah BTW, `/ci/` is also missing.

Comment (leemgs) [reply ci/doc/how-to-setup-taos-ci-server.md]: @dongju-chae , Sure,  For maintenance and readability.

Comment (leemgs) [reply ci/doc/how-to-setup-taos-ci-server.md]: > Ah BTW, /ci/ is also missing.

Nice catch. Fixed.

**Pull request merged**